test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -406,6 +426,15 @@ public void testNewConfigWithSwitchRole() throws Exception {
             replicationStatus.getSourceStatus().getReplicationInfo().getSnapshotSyncInfo().getStatus(),
             replicationStatus.getSourceStatus().getReplicationInfo().getSnapshotSyncInfo().getCompletedTime());
 
+        // Wait until data is fully replicated again
+        waitForReplication(size -> size == thirdBatch, mapSource, thirdBatch);
+        log.info("Data is fully replicated again after role switch, both maps have size {}. " +
+                        "Current source corfu[{}] log tail is {}, sink corfu[{}] log tail is {}",
+                thirdBatch, sourceClusterCorfuPort, sourceRuntime.getAddressSpaceView().getLogTail(),
+                sinkClusterCorfuPort, sinkRuntime.getAddressSpaceView().getLogTail());
+
+        TimeUnit.SECONDS.sleep(shortInterval);
```

Comment:
can we use a listener instead?

Reply:
This class is a bit old and it widely used sleep to wait for certain conditions, and I was reordering some code here to avoid checking status before data is actually replicated.

I think listener might not be suitable here as Source side could have updated the status table multiple times with state transition and periodic task. Maybe we can just block the assertion (like in waitForReplication) to wait until expected state reached. Can we address it in a separate PR to change all the places in this IT?